deploy/openshift.template.yaml

```diff
@@ -33,15 +32,18 @@ objects:
           app: git-partition-sync-producer
       spec:
         serviceAccountName: git-partition-sync-producer
-        volumes:
-        - name: private-gpg-key
-          secret:
-            secretName: private-gpg-key
         containers:
         - image: ${IMAGE}:${IMAGE_TAG}
           imagePullPolicy: Always
           name: git-partition-sync-producer
+          args: ["-dry-run=${DRY_RUN}"]
```

Comment:
Do you have another example of this in our codebase?

```The command and arguments that you define in the configuration file override the default command and arguments provided by the container image. If you define args, but do not define a command, the default command is used with your new arguments.

Note: The command field corresponds to entrypoint in some container runtimes.
```

https://kubernetes.io/docs/tasks/inject-data-application/define-command-argument-container/

Reply:
Entrypoint looks correct w/ vault-manager https://github.com/app-sre/vault-manager/blob/master/Dockerfile#L9

Reply:
I don't think curlies are correct.

`Note: The environment variable appears in parentheses, "$(VAR)". This is required for the variable to be expanded in the command or args field.`

https://kubernetes.io/docs/tasks/inject-data-application/define-command-argument-container/#use-environment-variables-to-define-arguments

Reply:
Dry run should be set from pod env vars, right? so `( )` over `{ }`, no?

Reply:
Opting for [aws-resource-exporter](https://github.com/app-sre/aws-resource-exporter/blob/master/Dockerfile#L10) entry point because that seems to be where some newer golang standards have been implemented.

Reply:
dry run is passed as a flag instead of set as environment variable

Reply:
I'm not confident vault manager's approach to command/args is necessary. If I interpret it correct, `command: ["/bin/sh"]` is overriding the default entry point (`/vault-manager`). This requires the additional args to execute the binary from `sh`.

I believe this can be simplified to omitting `command` (use default entrypoint) and just pass the single flag

Reply:
Yeah, I'm following the default entry point override.

Dry-run is set at the pod level though, right? You control that with an env var to the pod. Unless this is permanent dry run?

Reply:
Oh no, you mean is the setting in the SaaS file which then is templated into that variable. This is OK.

Reply:
For passing the flag, I was attempting to mirror how vault-manager performs it here: https://github.com/app-sre/vault-manager/blob/master/openshift/vault-manager.template.yaml#L43

Reply:
Unsure I'm answering the right question, but yes it would be permanent in the sense that whatever is set in the saas file for dry-run parameter will hold permanently for execution of specific pod

Reply:
Yeah I'm aligned with this